WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -492,6 +488,35 @@ private void updateConnectionBar(boolean isConnected) {
         }
     }
 
+    private void handleBlogRemoved() {
+        if (!AccountHelper.isSignedIn()) {
+            ActivityLauncher.showSignInForResult(this);
+        } else {
+            MySiteFragment mySiteFragment = getMySiteFragment();
+            if (mySiteFragment != null) {
+                int allBlogsCount = WordPress.wpDB.getAllBlogsCount();
+                if (allBlogsCount == 0) {
+                    mySiteFragment.setBlog(null);
+                } else {
+                    showSitePickerWithCurrentBlog();
+                }
+            }
+        }
+    }
+
+    private void showSitePickerWithCurrentBlog() {
+        Blog currentBlog = WordPress.getCurrentBlog();
+
+        int currentBlogId;
+        if (currentBlog != null) {
+            currentBlogId = currentBlog.getLocalTableBlogId();
+        } else {
+            currentBlogId = WordPress.wpDB.getAllBlogsIDs().get(0);
```

Comment:
This also isn't optimal since it's retrieving all blog IDs just to get the first one (plus it doesn't protect against an OOB exception). Maybe add something like this to WordPressDB instead (untested):
```
public int getFirstBlogId() {
    return SqlUtils.intForQuery(db, "SELECT id FROM " + BLOGS_TABLE + " LIMIT 1", null);
}
```